chain/chain/src/state_sync/adapter.rs

```diff
@@ -524,23 +527,26 @@ impl ChainStateSyncAdapter {
         shard_id: ShardId,
         sync_hash: CryptoHash,
         part_id: PartId,
-        data: &[u8],
+        part: &StatePart,
     ) -> Result<(), Error> {
         let shard_state_header = self.get_state_header(shard_id, sync_hash)?;
         let chunk = shard_state_header.take_chunk();
         let state_root = *chunk.take_header().take_inner().prev_state_root();
-        if !self.runtime_adapter.validate_state_part(shard_id, &state_root, part_id, data) {
+        if !self.runtime_adapter.validate_state_part(shard_id, &state_root, part_id, part) {
             byzantine_assert!(false);
             return Err(Error::Other(format!(
                 "set_state_part failed: validate_state_part failed. state_root={:?}",
                 state_root
             )));
         }
+        let epoch_id = self.epoch_manager.get_epoch_id_from_prev_block(&sync_hash)?;
+        let protocol_version = self.epoch_manager.get_epoch_protocol_version(&epoch_id)?;
```

Comment:
Sanity check - why is sync hash the previous block here?

Reply:
`sync_hash` is close to the beginning of epoch, but you are right, the less assumptions the better, fixed that